crates/matrix-sdk/src/room/timeline/inner.rs

```diff
@@ -321,52 +325,59 @@ impl<P: ProfileProvider> TimelineInner<P> {
             }
         };
 
-        let maybe_retry_decrypt = |item: Arc<TimelineItem>| async move {
-            let event_item = item.as_event()?;
-
-            match event_item.content().as_unable_to_decrypt()? {
-                EncryptedMessage::MegolmV1AesSha2 { session_id, .. }
-                    if should_retry(session_id) =>
-                {
-                    let EventTimelineItem::Remote(
-                        RemoteEventTimelineItem { event_id, raw, .. },
-                    ) = event_item else {
-                        error!("Key for unable-to-decrypt timeline item is not an event ID");
-                        return None;
-                    };
-
-                    let event = match olm_machine.decrypt_room_event(raw.cast_ref(), room_id).await
+        let retry_one = |item: Arc<TimelineItem>| {
```

Comment:
I wonder if we should put this into a separate method, instead of having a closure here, though not that important, if you disagree feel free to ignore the comment.